_data-prepper/pipelines/configuration/processors/otel-trace-group.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: default
+title: OTel trace group
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# OTel trace group processor
+
+The OpenTelemetry (OTel) trace group processor completes missing trace group related fields in the collection of [span](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/trace/Span.java) records by looking up the OpenSearch backend. The OTel trace group processor identifies the missing trace group information for a `spanId` by looking up the relevant fields in its root `span` stored in OpenSearch.
+
+## OpenSearch
+
+See the following example `YAML` configuration file for the OTel trace group processor when you connect to an OpenSearch cluster using your username and password:
+
+``` YAML
+pipeline:
+  ...
+  processor:
+    - otel_trace_group:
+        hosts: ["https://localhost:9200"]
+        cert: path/to/cert
+        username: YOUR_USERNAME_HERE
+        password: YOUR_PASSWORD_HERE
+```
+
+See [OpenSearch security](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-plugins/opensearch/opensearch_security.md#L4) for a more detailed explanation of which OpenSearch credentials and permissions are required and how to configure those credentials for the OTel trace group processor.
```

Comment:
If my PR (https://github.com/opensearch-project/documentation-website/pull/3809) gets merged before this, you can update this link to point to my security docs.

Reply:
Okay, can you now link off to here? https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sinks/opensearch/#amazon-opensearch-service-domain-security

Reply:
@lizsnyder Updated this link - thanks!